Code/Legacy/CrySystem/XConsole.cpp

```diff
@@ -2585,57 +2534,44 @@ void CXConsole::RemoveOutputPrintSink(IOutputPrintSink* inpSink)
 
 
 //////////////////////////////////////////////////////////////////////////
-void CXConsole::AddLinePlus(const char* inputStr)
+void CXConsole::AddLinePlus(AZStd::string_view inputStr)
 {
-    AZStd::string str, tmpStr;
-
+    if (m_dqConsoleBuffer.empty())
     {
-        if (!m_dqConsoleBuffer.size())
-        {
-            return;
-        }
-
-        str = inputStr;
-
-        // strip trailing \n or \r.
-        if (!str.empty() && (str[str.size() - 1] == '\n' || str[str.size() - 1] == '\r'))
-        {
-            str.resize(str.size() - 1);
-        }
+        return;
+    }
 
-        AZStd::string::size_type nPos;
-        while ((nPos = str.find('\n')) != AZStd::string::npos)
+    AZStd::string str;
+    auto ParseLine = [&str, firstIteration = true](AZStd::string_view line) mutable
+    {
+        // Add <space> between lines
+        if (!firstIteration)
         {
-            str.replace(nPos, 1, 1, ' ');
+            str += ' ';
         }
+        firstIteration = false;
 
-        while ((nPos = str.find('\r')) != AZStd::string::npos)
-        {
-            str.replace(nPos, 1, 1, ' ');
-        }
+        str += line;
+    };
+    AZ::StringFunc::TokenizeVisitor(inputStr, ParseLine, "\r\n");
 
-        tmpStr = m_dqConsoleBuffer.back();// += str;
+    AZStd::string tmpStr = m_dqConsoleBuffer.back();
 
-        m_dqConsoleBuffer.pop_back();
+    m_dqConsoleBuffer.pop_back();
 
-        if (tmpStr.size() < 256)
-        {
-            m_dqConsoleBuffer.push_back(tmpStr + str);
-        }
-        else
-        {
-            m_dqConsoleBuffer.push_back(tmpStr);
-        }
+    if (tmpStr.size() < 256)
```

Comment:
question: Is it worth giving this magic number a name and making it a constant?

Reply:
I honestly don't know what the 256 is for and why if the temporary string is less than that, that it appends the other string to it.

The `m_dqConsoleBuffer` AZStd::deque is strange in the first place.
It seems to cache of strings that are less than the `con_line_buffer_size` console variable, but it doesn't seem like there is a need to do that.

Reply:
Lol okay, yeah it is pretty confusing. It's fine to leave as-is I just wanted to check if we could improve it for future maintainers

Reply:
I am thinking about just remove it after looking into it a bit more.

Reply:
Nice! Good plan 🙂